processing/src/main/java/io/druid/segment/filter/LikeFilter.java

```diff
@@ -0,0 +1,146 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment.filter;
+
+import com.google.common.base.Strings;
+import com.metamx.collections.bitmap.ImmutableBitmap;
+import io.druid.query.extraction.ExtractionFn;
+import io.druid.query.filter.BitmapIndexSelector;
+import io.druid.query.filter.Filter;
+import io.druid.query.filter.LikeDimFilter;
+import io.druid.query.filter.ValueMatcher;
+import io.druid.query.filter.ValueMatcherFactory;
+import io.druid.segment.column.BitmapIndex;
+import io.druid.segment.data.Indexed;
+
+import java.util.Iterator;
+
+public class LikeFilter implements Filter
+{
+  private static final String HIGHEST_CHARACTER = String.valueOf(Character.MAX_VALUE);
+
+  private final String dimension;
+  private final ExtractionFn extractionFn;
+  private final LikeDimFilter.LikeMatcher likeMatcher;
+
+  public LikeFilter(
+      final String dimension,
+      final ExtractionFn extractionFn,
+      final LikeDimFilter.LikeMatcher likeMatcher
+  )
+  {
+    this.dimension = dimension;
+    this.extractionFn = extractionFn;
+    this.likeMatcher = likeMatcher;
+  }
+
+  @Override
+  public ImmutableBitmap getBitmapIndex(BitmapIndexSelector selector)
+  {
+    if (extractionFn == null && likeMatcher.getSuffixStyle() == LikeDimFilter.LikeMatcher.SuffixStyle.MATCH_EMPTY) {
+      // dimension equals prefix
+      return selector.getBitmapIndex(dimension, likeMatcher.getPrefix());
+    } else if (extractionFn == null) {
+      // dimension startsWith prefix (and maybe matches pattern)
+      final BitmapIndex bitmapIndex = selector.getBitmapIndex(dimension);
+
+      if (bitmapIndex == null) {
+        // Treat this as a column full of nulls
+        return likeMatcher.matches(null) ? Filters.allTrue(selector) : Filters.allFalse(selector);
+      }
+
+      // search for start, end indexes in the bitmaps; then include all matching bitmaps between those points
+      final Indexed<String> dimValues = selector.getDimensionValues(dimension);
+
+      final String lower = Strings.nullToEmpty(likeMatcher.getPrefix());
+      final String upper = Strings.nullToEmpty(likeMatcher.getPrefix()) + HIGHEST_CHARACTER;
+      final int startIndex; // inclusive
+      final int endIndex; // exclusive
+
+      final int lowerFound = bitmapIndex.getIndex(lower);
+      startIndex = lowerFound >= 0 ? lowerFound : -(lowerFound + 1);
+
+      final int upperFound = bitmapIndex.getIndex(upper);
+      endIndex = upperFound >= 0 ? upperFound + 1 : -(upperFound + 1);
+
+      // Union bitmaps for all matching dimension values in range
+      return selector.getBitmapFactory().union(
+          new Iterable<ImmutableBitmap>()
```

Comment:
This `Iterable` chain of nested classes is very noisy. Maybe create a simple List of bitmaps, fill it in a for loop with i between startIndex and endIndex, then return `bitmapFactory.union(bitmaps)`?